### PR TITLE
Type

### PR DESCRIPTION
**Pull request by Alizter**
as discussed in #132

### CODE REVIEW

**Comment by Alizter** [comment]
hmm I may have messed things up a bit here

**Comment by mortberg** [comment]
> hmm I may have messed things up a bit here

What do you mean?

File: Cubical/Data/Fin/Base.agda [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+open import Cubical.Relation.Nullary
+
+-- Finite types.
+--
+-- Currently it is most convenient to define these as a subtype of the
+-- natural numbers, because indexed inductive definitions don't behave
+-- well with cubical Agda. This definition also has some more general
+-- attractive properties, of course, such as easy conversion back to
+-- ℕ.
+Fin : ℕ → Set
```
Comment (mortberg) [inline Cubical/Data/Fin/Base.agda]: Set?

File: Cubical/Data/Fin/Base.agda [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+fsplit (0 , k<sn) = inl (toℕ-injective refl)
+fsplit (suc k , k<sn) = inr ((k , pred-≤-pred k<sn) , toℕ-injective refl)
+
+-- Fin 0 is empty
+¬Fin0 : ¬ Fin 0
+¬Fin0 (k , k<0) = ¬-<-zero k<0
+
+-- The full inductive family eliminator for finite types.
+finduction
+  : ∀(P : ∀{k} → Fin k → Set ℓ)
```
Comment (mortberg) [inline Cubical/Data/Fin/Base.agda]: Set

File: Cubical/Data/Fin/Properties.agda [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+open import Cubical.Data.Empty
+open import Cubical.Data.Sum
+open import Cubical.Data.Sigma
+
+open import Cubical.Induction.WellFounded
+
+private
+  -- Σ is more convenient than the inductive ×, but I don't
+  -- want to have to write λ_→ all over.
+  _×_ : Set → Set → Set
```
Comment (mortberg) [inline Cubical/Data/Fin/Properties.agda]: Set

File: Cubical/Data/Fin/Properties.agda [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    ∘ inj-m+ {suc k}
+
+  expand×Emb : ∀ k → isEmbedding (expand× {k})
+  expand×Emb 0 = ⊥-elim ∘ ¬Fin0 ∘ fst
+  expand×Emb (suc k)
+    = injEmbedding (isOfHLevelΣ 2 isSetFin (λ _ → isSetℕ)) isSetℕ (expand×Inj k)
+
+-- A Residue is a family of types representing evidence that a
+-- natural is congruent to a value of a finite type.
+Residue : ℕ → ℕ → Set
```
Comment (mortberg) [inline Cubical/Data/Fin/Properties.agda]: Set

**Comment by mortberg** [comment]
It doesn't seem like you changed all of the Set to Type

**Comment by mortberg** [comment]
I rebased my PR on top of master now. Hopefully you can get the diff to look correct

**Comment by mortberg** [comment]
Actually, I'll just quickly do the necessary changes on the other PR so that we can merge it. So I'll close this one now

**Pull request closed without merging**